pymc3/sampling.py

```diff
@@ -1207,6 +1207,63 @@ def sample_ppc_w(traces, samples=None, models=None, weights=None,
     return {k: np.asarray(v) for k, v in ppc.items()}
 
 
+def sample_prior(samples=500, model=None, vars=None, size=None,
+                   random_seed=None, progressbar=True):
+    """Generate samples from the prior of a model.
+
+    Parameters
+    ----------
+    samples : int
+        Number of samples from the prior to generate. Defaults to 500.
+    model : Model (optional if in `with` context)
+    vars : iterable
+        Variables for which to compute the posterior predictive samples.
+        Defaults to `model.named_vars`.
+    size : int
+        The number of random draws from the distribution specified by the
+        parameters in each sample of the trace.
+    random_seed : int
+        Seed for the random number generator.
+    progressbar : bool
+        Whether or not to display a progress bar in the command line.
+
+    Returns
+    -------
+    dict
+        Dictionary with the variables as keys. The values are arrays of prior samples.
+    """
+        
+    model = modelcontext(model)
+
+    if vars is None:
+        vars = set(model.named_vars.keys())
+
+    if random_seed is not None:
+        np.random.seed(random_seed)
+
+    if progressbar:
+        indices = tqdm(range(samples))
+
+    try:
+        prior = {var: [] for var in vars}
+        for _ in indices:
+            point = {}
+            for var_name, var in model.named_vars.items():
+                val = var.distribution.random(point=point, size=size)
```

Comment:
It is better to use the part from `smc` where it samples from the prior:
https://github.com/pymc-devs/pymc3/blob/801accb5f236ab9daa89a8fcd9d09a3ba4ed0a39/pymc3/step_methods/smc.py#L186-L193
Otherwise you will get error with bounded RVs:
`AttributeError: 'TransformedDistribution' object has no attribute 'random'`